internal/services/containers/kubernetes_cluster_validate.go

```diff
@@ -26,14 +26,16 @@ func validateKubernetesCluster(d *pluginsdk.ResourceData, cluster *containerserv
 				dnsServiceIP := profile["dns_service_ip"].(string)
 				serviceCidr := profile["service_cidr"].(string)
 				podCidr := profile["pod_cidr"].(string)
+				serviceCidrs := profile["service_cidrs"].([]interface{})
+				isCidrset := serviceCidr != "" || len(serviceCidrs) != 0
 
 				// Azure network plugin is not compatible with pod_cidr
 				if podCidr != "" && networkPlugin == "azure" {
 					return fmt.Errorf("`pod_cidr` and `azure` cannot be set together")
 				}
 
 				// if not All empty values or All set values.
-				if !(dockerBridgeCidr == "" && dnsServiceIP == "" && serviceCidr == "") && !(dockerBridgeCidr != "" && dnsServiceIP != "" && serviceCidr != "") {
+				if !(dockerBridgeCidr == "" && dnsServiceIP == "" && !isCidrset) && !(dockerBridgeCidr != "" && dnsServiceIP != "" && isCidrset) {
```

Comment:
This validation makes sure that `docker_bridge_cidr`, `dns_service_ip` and `service_cidr` are either all set or empty rather than preventing the user from setting both `service_cidr`/`service_cidrs` and `pod_cidr`/`pod_cidrs`. In any case after trying this out it is indeed possible to set both `_cidr` and `_cidrs`.

However it seems this validation may no longer be relevant or needed since AKS no longer supports docker for the container runtime (see #18119). Could you please update the validation here while you're at it?

Reply:
About depreciating `docker_bridge_cidr`, I think it's better to make another PR for it.